[DOC] updates to oauth config chapter including latest config attributes (#2313)
strimzi/strimzi-kafka-operator: documentation/modules/oauth/proc-oauth-kafka-config.adoc

. Create or edit the resource for the Kafka component so that {oauth} authentication is configured for the authentication property.
+
For {oauth} authentication, you can use:
+
--
* Client ID and secret
* Client ID and refresh token
* Access token
* TLS
--
+
xref:appendix_crds#type-KafkaClientAuthenticationOAuth-reference[KafkaClientAuthenticationOAuth schema reference provides examples of each].
+
For example, here {oauth} is assigned to the Kafka Bridge client using a client ID and secret, and TLS:
+
[source,yaml,subs="+quotes,attributes"]
----
apiVersion: {KafkaApiVersion}
kind: KafkaBridge
metadata:
  name: my-bridge
spec:
  # ...
  authentication:
    type: oauth <1>
    tokenEndpointUri: https://<auth-server-address>/auth/realms/master/protocol/openid-connect/token <2>
    clientId: kafka-bridge
    clientSecret:
      secretName: my-bridge-oauth
      key: clientSecret
    tlsTrustedCertificates: <3>
      - secretName: oauth-server-cert

>       certificate: tls.crt

Reviewer: I think the spacing is wrong here:

```suggestion
        certificate: tls.crt
```

----
<1> Authentication type set to `oauth`.
<2> URI of the token endpoint for authentication.
<3> Trusted certificates for TLS connection to the authorization server.
+

> Depending on how you applying {oauth} authentication, and the type of authorization server, there are additional configuration options you can use:

Reviewer 2: ```suggestion
Depending on how you apply {oauth} authentication, and the type of authorization server, there are additional configuration options you can use:
```

+
[source,yaml,subs="+quotes,attributes"]
----
# ...
spec:
  # ...
  authentication:
    # ...
    disableTlsHostnameVerification: true <1>

>     skipTypeCheck: true <2>

Reviewer: ```suggestion
    checkAccessTokenType: false <2>
```

Author: Single change in doc for this one.

>     notJwt: true <3>

Reviewer: ```suggestion
    accessTokenIsJwt: false <3>
```

Author: Two changes in doc

----
<1> (Optional) Disable TLS hostname verification. Default is `false`.

> <2> If the authorization server does not return a `typ` (type) claim inside the JWT token, you can skip the token type check. Default is `false`.

Reviewer: ```suggestion
<2> If the authorization server does not return a `typ` (type) claim inside the JWT token, you can skip the token type check. Default is `true`.
```

> <3> If you are using opaque tokens, you can apply `notJwt: true` so that access tokens are not treated as JWT tokens.

Reviewer: ```suggestion
<3> If you are using opaque tokens, you can apply `accessTokenIsJwt: false` so that access tokens are not treated as JWT tokens.
```
